www/base/src/app/layout.jade

```diff
@@ -22,3 +22,11 @@ html.no-js(xmlns:ng='http://angularjs.org', xmlns:app='ignored')
     | {% endfor %}
     script
       | angular.module("app").constant("config", {{configjson|safe}})
+    | {% if custom_templates %}
+    script
+      | angular.module("app").run(function ($templateCache) {
+      | {% for template_name, content in custom_templates.items() %}
+      |  $templateCache.put("{{ template_name }}", {{ content }});
+      | {% endfor %}
+      | })
+    | {% endif %}
```

Comment:
see original code: the | safe was important for me when I implemented this feature.
```
+    script
+        | angular.module("app").run(function($templateCache){
+        | {% for name, html in custom_templates.items() -%}
+        |     $templateCache.put("{{name}}", {{html|safe}})
+        | {%- endfor -%}
+        | })
```